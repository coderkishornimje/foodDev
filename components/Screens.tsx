import {StackView, createStackNavigator} from '@react-navigation/stack';

import {Text, StyleSheet, View} from 'react-native';
import React, {Component} from 'react';
import OnboardScreen from './OnboardScreen';
import Login from './Login';
import Splash from './Splash';
import Register from './Register';
import OtpVerify from './OtpVerify';
import AllowLocation from './AllowLocation';
import Home from './Home';
import CustomTabBar from './CustomTabBar';
import Offer from './Offer';
import TabNavigator from './TabNavigator';
import Search from './Search';
import Cart from './Cart';
import Notification from './Notification';
import Main from './profile/Main';
import {Tab} from 'react-native-elements';
import TabContainer from './TabContainer';
import Favorite from './profile/Favorite';
import Editprofile from './profile/Editprofile';
import Order from './profile/Order';
import ViewAllTodayFood from './ViewAllTodayFood';
import AllPizaas from './AllPizaas';
import ShowFoods from './food-category/ShowFoods';
import NearbyResto from './NearbyResto';

const Stack = createStackNavigator();

export default class Screens extends Component {
  render() {
    return (
      <Stack.Navigator
        initialRouteName="OnboardScreen"
        screenOptions={{headerShown: false}}>
        <Stack.Screen name="OnboardScreen" component={OnboardScreen} />
        <Stack.Screen name="Login" component={Login} />
        <Stack.Screen name="Splash" component={Splash} />
        <Stack.Screen name="Register" component={Register} />
        <Stack.Screen name="OtpVerify" component={OtpVerify} />
        <Stack.Screen name="AllowLocation" component={AllowLocation} />
        
        <Stack.Screen name="Home" component={Home} />
        {/* <Stack.Screen name="CustomTabBar" component={CustomTabBar} /> */}
        <Stack.Screen name="Offer" component={Offer} />
        <Stack.Screen name="Search" component={Search} />
        <Stack.Screen name="Cart" component={Cart} />
        <Stack.Screen name="Notification" component={Notification} />
        <Stack.Screen name="ViewAllTodayFood" component={ViewAllTodayFood} />
        <Stack.Screen name="AllPizaas" component={AllPizaas} />
        <Stack.Screen name="ShowFoods" component={ShowFoods} />
        <Stack.Screen name="NearbyResto" component={NearbyResto} />
        

        <Stack.Screen name="Main" component={Main} />
        <Stack.Screen name="Favorite" component={Favorite} />
        <Stack.Screen name="Editprofile" component={Editprofile} />
        <Stack.Screen name="Order" component={Order} />
        
        <Stack.Screen name="TabContainer" component={TabContainer} />
      </Stack.Navigator>
    );
  }
}

const styles = StyleSheet.create({});
