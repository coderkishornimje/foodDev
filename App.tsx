import 'react-native-gesture-handler';
import {Text, StyleSheet, View,Platform} from 'react-native';
import React, {Component} from 'react';
import Splash from './components/Splash';
import {NavigationContainer} from '@react-navigation/native';
import {Screen} from 'react-native-screens';
import Screens from './components/Screens';
import CustomTabBar from './components/CustomTabBar';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import AppContextProvider from './components/AppContext';

interface IState {
  isLoading: boolean;
}
interface IProps {
  navigation?: any;
}
const Tab = createBottomTabNavigator();
export default class App extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      isLoading: true,
    };
  }

  componentDidMount(): void {
    setTimeout(() => {
      this.setState({isLoading: false});
    }, 2000);
  }

  render() {
    if (this.state.isLoading) {
      return <Splash />;
    }

    return (
      <AppContextProvider>
      <NavigationContainer>
         <Screens/>
      </NavigationContainer>
      </AppContextProvider>
    );
  }
}

const styles = StyleSheet.create({});
