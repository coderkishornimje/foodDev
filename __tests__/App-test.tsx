import 'react-native-gesture-handler/jestSetup';
import 'react-native';
import React from 'react';
import App from '../App';
import {cleanup, fireEvent, render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Screens from '../components/Screens';

jest.mock('react-native-phone-number-input', () => 'PhoneInput');
jest.mock('@twotalltotems/react-native-otp-input',()=>"OTPInputView")


describe('app component', () => {
  it('renders correctly', () => {
    renderer.create(<App />);
  });

  test('to check app render correctly', () => {
    const appRender = render(<App />);
    expect(appRender).toMatchSnapshot();
    expect(appRender).toBeDefined();
  });
  
  test('screen contains a button linking to the notifications page', async () => {
    const {getByTestId} = render(<App />);
    const component = (
      <NavigationContainer>
        <Screens/>
      </NavigationContainer>
    );
    render(component);
  });
  
  test('to check settimeout',()=>{
    jest.useFakeTimers();
    const {getByTestId} = render(<App />);
     jest.advanceTimersByTime(2000);
  })
  
});

