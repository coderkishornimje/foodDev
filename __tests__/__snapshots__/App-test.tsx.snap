// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component to check app render correctly 1`] = `
<View
  style={
    {
      "height": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="stretch"
      source={
        {
          "testUri": "../../../images/Rectangle.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "bottom": 100,
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "right": 100,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../images/Sicon.png",
          }
        }
        style={
          {
            "height": 100,
            "width": 200,
          }
        }
      />
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 35,
            "fontWeight": "bold",
            "marginLeft": 45,
          }
        }
      >
        Food Delivery
      </Text>
    </View>
  </View>
</View>
`;
